### DISCUSSION TITLE
Unable to use pyFAST in WSL2

### DISCUSSION THREAD

**Discussion by general-rishkin** [FAST-Imaging/FAST#217] [Q&A] [answered]
I am trying to download the data using the frist python tutorial, i.e., 

```
import fast
fast.downloadTestDataIfNotExists()
```
but I get the error, `Unhandled exception caught during execution: clGetPlatformIDs`.

Is there a way to make this use CUDA or bypass the clGetPlatformIDs call?

**Comment by smistad** [comment, marked as answer]
Hi @general-rishkin 

I have never tried running FAST in WSL. The error you get most likely means that OpenCL is not installed properly.
You could try using the POCL implementation, a user has reported that it works here: https://answers.microsoft.com/en-us/windows/forum/all/is-opencl-supported-on-wsl2/71e8eeee-14cc-49e4-852a-1827228ed1c4
This will most likely enable you to run FAST on the CPU, I am not sure how to get FAST to work on the GPU through WSL.

You can install POCL using the one that exists in apt on ubuntu (sudo apt install libpocl2), or use a newer one by building from github (https://github.com/pocl/pocl/ https://github.com/gyferlim/pocl/blob/main/ADDITIONAL.md).

Note that using WSL and POCL most likely will not give you the best performance. FAST runs fine on windows with pyFAST, so if you are interested in best performance I would advice you to use that instead.

Good luck and please let me know if it works out or not.

**Reply by general-rishkin** [reply]
Thanks @smistad. I will have a look at these although at the moment, I have turned my attention to running it on Linux.
